boards/arc/hsdk/hsdk_2cores_defconfig

```diff
@@ -12,6 +12,5 @@ CONFIG_CONSOLE=y
 CONFIG_UART_CONSOLE=y
 CONFIG_SERIAL=y
 CONFIG_GPIO=y
```

Comment:
I`m in favor to drop GPIO too. It became so essential that all boards must have it. In this case, I think it will be better define in the SoC with default=y. Intention is mimics same use case that some pinctrl already have been using in the project. Like this https://github.com/zephyrproject-rtos/zephyr/blob/8e55e59c5917b61d563d5e8bf759e8aeb746ff8b/soc/arm/gigadevice/Kconfig.defconfig#L8-L9

Reply:
that's a good point. I wonder if we should go this route for other things as well (e.g. MPU). Let's discuss this in architecture meeting @carlescufi.